pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1309,7 +1309,10 @@ func buildHTTPConnectionManager(listenerOpts buildListenerOpts, httpOpts *httpLi
 
 	accessLogBuilder.setHTTPAccessLog(listenerOpts.push.Mesh, connectionManager)
 
-	configureTracing(listenerOpts, connectionManager)
+	filterModifierCtx := &xdsfilters.FilterModifierContext{}
+	configureTracing(listenerOpts, connectionManager, filterModifierCtx)
+
+	xdsfilters.ModifyFilter(filterModifierCtx, connectionManager.HttpFilters)
```

Comment:
I would suggest mutating it in the function, so can get rid of additional marshal/unmarshal

Reply:
I mean in `configureTracing` ir even low level

Reply:
deleted `modifier.go`. changed to build router filter in-place. we don't have to take care marshaling anymore.